### PR TITLE
feat(trait): Move addon Telemetry to traits

### PR DESCRIPTION
**Pull request by gansheer**
Ref #5814

* Transform Telemetry addon to trait
* Ensure kamel CLI creates new integrations with telemetry as trait
* keep retrocompatibility with addon spec:
```
  traits:
    addons:
      telemetry:
        enabled: true
        endpoint: http://opentelemetrycollector.otlp:4317
```

**Release Note**
```release-note
feat(trait): Move addon Telemetry to traits
```

### CODE REVIEW

**Review by oscerd — APPROVED** [review]

**Review by squakez — COMMENTED** [review]
LGTM. I think you need to check how the `kamel run` would convert to an addons spec or to an trait spec.

File: pkg/cmd/run_test.go [hunk 1]
```diff
@@ -438,6 +438,7 @@ func TestConfigureTraits(t *testing.T) {
 		"--trait", "affinity.pod-affinity=false",
 		"--trait", "environment.container-meta=false",
 		"--trait", "prometheus.pod-monitor=false",
+		"--trait", "telemetry.auto=true",
```
Comment (gansheer) [inline pkg/cmd/run_test.go:441]: @squakez That was the point of this change to test is the run command really creates a trait and not an addon.

**Review by squakez — COMMENTED** [review]
What I miss is some deprecation notice in the Integration which is using the addons. Ideally, when executing the trait logic, it should be reported as an Integration condition as it happens for instance here: https://github.com/apache/camel-k/blob/a664283ef4148a40987380a44b917613fe17a314/pkg/trait/builder.go#L98

**Review by squakez — APPROVED** [review]

File: pkg/trait/telemetry.go [hunk 2]
```diff
@@ -84,36 +56,50 @@ var (
...
+func (t *telemetryTrait) Configure(e *Environment) (bool, *TraitCondition, error) {
 	if e.Integration == nil || !ptr.Deref(t.Enabled, false) {
 		return false, nil, nil
 	}
 
-	if !ptr.Deref(t.Auto, true) {
-		return true, nil, nil
+	var condition *TraitCondition
+
+	if _, isAddon := e.Integration.Spec.Traits.Addons["telemetry"]; isAddon {
```
Comment (squakez) [inline pkg/trait/telemetry.go:72]: Ah, we also need to add a `// Deprecated` comment as it's the way I usually scan code for deprecations to be eventually removed.

**Pull request merged**